packages/wonder-blocks-button/components/clickable-behavior.js

```diff
@@ -0,0 +1,190 @@
+    onMouseLeave: () => void,
+    onMouseDown: () => void,
+    onMouseUp: () => void,
+    onTouchStart: () => void,
+    onTouchEnd: () => void,
+    onKeyDown: (e: KeyboardEvent) => void,
+    onKeyUp: (e: KeyboardEvent) => void,
+    onBlur: () => void,
+};
+
+const keyCodes = {
+    tab: 9,
+    enter: 13,
+    space: 32,
+};
+
+export default class ClickableBehavior extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            hovered: false,
+            focused: false,
+            pressed: false,
+        };
+    }
+
+    waitingForClick: boolean;
+
+    handleClick = (e: SyntheticMouseEvent<>) => {
+        if (!this.props.disabled && this.props.onClick) {
+            this.waitingForClick = false;
+            this.props.onClick(e);
+        }
+    };
+
+    handleMouseEnter = () => {
+        if (!this.props.disabled && !this.waitingForClick) {
+            this.setState({hovered: true});
+        }
+    };
+
+    handleMouseLeave = () => {
+        if (!this.props.disabled && !this.waitingForClick) {
+            // TODO(yejia): Is this what we want for pressed?
+            this.setState({hovered: false, pressed: false});
+        }
+    };
+
+    handleMouseDown = () => {
+        if (!this.props.disabled) {
+            this.setState({pressed: true});
+        }
+    };
+
+    handleMouseUp = () => {
+        if (!this.props.disabled) {
+            this.setState({pressed: false});
+        }
+    };
+
+    handleTouchStart = () => {
+        if (!this.props.disabled) {
+            this.setState({pressed: true});
+        }
+    };
+
+    handleTouchEnd = () => {
+        if (!this.props.disabled) {
+            this.setState({pressed: false});
+            this.waitingForClick = true;
+        }
+    };
+
+    handleKeyDown = (event: KeyboardEvent) => {
+        const keyCode = event.which || event.keyCode;
+        const {disabled, href} = this.props;
+        if (!disabled) {
+            if (keyCode === keyCodes.tab) {
+                this.setState({focused: false});
+            } else if (
+                href ? keyCode === keyCodes.enter : keyCode === keyCodes.space
+            ) {
+                this.setState({pressed: true});
+            }
+        }
+    };
+
+    handleKeyUp = (event: KeyboardEvent) => {
+        const keyCode = event.which || event.keyCode;
+        const {disabled, href} = this.props;
+        if (!disabled) {
+            if (keyCode === keyCodes.tab) {
+                this.setState({focused: true});
+            } else if (
+                href ? keyCode === keyCodes.enter : keyCode === keyCodes.space
+            ) {
+                this.setState({pressed: false});
+            }
+        }
+    };
+
+    handleBlur = () => {
+        if (!this.props.disabled) {
+            this.setState({focused: false, pressed: false});
+        }
+    };
+
+    render() {
+        const handlers = {
```

Comment:
Every single one of these functions has `if (!this.props.disabled...)` in it. It seems like it would be easier to do something like: `const handlers = disabled ? {} : { onClick: ..., etc. };`. Or you could stub out empty functions for each of them, make it a constant at the top of the file, and just pass that in instead. That will make all your handler functions a lot easier to read!